src/internal/operations/restore.go

```diff
@@ -228,6 +229,25 @@ func (op *RestoreOperation) do(
 		"restore_protected_resource_id", restoreToProtectedResource.ID(),
 		"restore_protected_resource_name", clues.Hide(restoreToProtectedResource.Name()))
 
+	// Check if the resource has the service enabled to be able to restore.
+	runnable, err := op.rc.IsServiceEnabled(
+		ctx,
+		op.Selectors.PathService(),
+		restoreToProtectedResource.ID())
+	if err != nil {
+		logger.CtxErr(ctx, err).Error("verifying restore is runnable")
+		op.Errors.Fail(clues.Wrap(err, "verifying restore is runnable"))
```

Comment:
The `run()` func that calls this already establishes `errors.Fail` when an error is returned.  This line should be removed.